agent/replacecall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -0,0 +1,398 @@
+  private final Set<String> excludedPackagePrefixes;
+
+  /**
+   * Create a {@link CallReplacementTransformer} that transforms method calls in classes other than
+   * those named in the given exclusion set.
+   *
+   * <p>The transformer can be run by multiple threads, so the replacement maps use concurrent
+   * implementations.
+   *
+   * @param replacementMap the concurrent hash map with method replacements
+   * @param excludedPackagePrefixes the period-terminated prefixes for packages from which classes
+   *     should not be transformed
+   */
+  CallReplacementTransformer(
+      ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap,
+      Set<String> excludedPackagePrefixes) {
+    this.replacementMap = replacementMap;
+    this.excludedPackagePrefixes = excludedPackagePrefixes;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Transforms the given class class by replacing calls to methods with corresponding calls as
+   * determined by {@link #replacementMap}.
+   *
+   * <p>Excludes bootloaded classes that are not AWT/Swing classes. Other exclusions are determined
+   * by the set of {@link #excludedPackagePrefixes}.
+   *
+   * @see ReplaceCallAgent
+   */
+  @Override
+  public byte[] transform(
+      ClassLoader loader,
+      String className,
+      Class<?> classBeingRedefined,
+      ProtectionDomain protectionDomain,
+      byte[] classfileBuffer)
+      throws IllegalClassFormatException {
+    // Note: An uncaught exception within a transform method results in null being returned.
+
+    String fullClassName = className.replace("/", ".");
+
+    if (isExcludedClass(loader, fullClassName)) {
+      debug_transform.log("transform: ignoring excluded class %s%n", className);
+      return null;
+    }
+
+    debug_transform.log("%ntransform: ENTER %s%n", className);
+
+    // Parse the bytes of the classfile
+    JavaClass c;
+    try {
+      ClassParser parser = new ClassParser(new ByteArrayInputStream(classfileBuffer), className);
+      c = parser.parse();
+    } catch (Exception e) {
+      debug_transform.log("transform: EXIT parse of %s resulted in error %s%n", className, e);
+      throw new RuntimeException("Unexpected parse exception", e);
+    }
+
+    try {
+      ClassGen cg = new ClassGen(c);
+      if (transformClass(cg)) {
+        JavaClass javaClass = cg.getJavaClass();
+        if (ReplaceCallAgent.debug) {
+          Path filepath = ReplaceCallAgent.debugPath.resolve(className + ".class");
+          javaClass.dump(filepath.toFile());
+        }
+        debug_transform.log("transform: EXIT class %s transformed", className);
+        return javaClass.getBytes();
+      } else {
+        debug_transform.log("transform: EXIT class %s not transformed", className);
+        return null;
+      }
+    } catch (Throwable e) {
+      debug_transform.log(
+          "transform: EXIT transform of %s resulted in exception %s%n", className, e);
```

Comment:
This particular log traces the agent as it enters/exits each method.  Adding the stack trace won't add any information.